src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const UsedSeparator = separator;
+
+    for (let key = 0; key < this.length; key++) {
+      if (key === this.length - 1) {
+        if (!(this[key] === undefined || this[key] === null)) {
+          result = result + this[key];
+        };
+        break;
+      }
+
+      if (this[key] === undefined || this[key] === null) {
+        result = result + `${UsedSeparator}`;
+      } else {
+        result = result + this[key] + `${UsedSeparator}`;
+      }
+    }
+
+    return result;
```

Comment:
```suggestion
   for (let i = 0; i < this.length; i++) {
      if (this[i] === null || this[i] === undefined) {
        this[i] = '';
      }

      if (i === this.length - 1) {
        result += this[i];
      } else {
        result += this[i];
        result += separator;
      }
    }

    return result;
```

давай максимально спростимо, і не треба використовувати вкладеність умов if